### PR TITLE
feat: logging query and data in Unmarshall method

### PR DESCRIPTION
**Pull request by saminahbab**
### What

Change the place that the cantabular client logs query data (one level up)  for failed cantabular queries

### How to review

look at changes 

### Who can review

Team B

### CODE REVIEW

File: cantabular/queries.go [hunk 1]
```diff
@@ -279,20 +285,15 @@ func (c *Client) queryUnmarshal(ctx context.Context, graphQLQuery string, data Q
...
 			res.StatusCode,
-			log.Data{
-				"url": url,
-			},
+			logData,
 		)
 	}
 
 	if err := json.Unmarshal(b, v); err != nil {
 		return dperrors.New(
```
Comment (ChrysmOre) [inline cantabular/queries.go:293]: Notice this error has the extra responsebody.String() added, you can add it to the logData here

File: cantabular/queries.go [hunk 2]
```diff
@@ -268,31 +268,32 @@ func (data *QueryData) Encode(query string) (bytes.Buffer, error) {
...
+	logData := log.Data{
+		"url":        url,
+		"query":      graphQLQuery,
+		"query_data": data,
+	}
+
 	res, err := c.postQuery(ctx, graphQLQuery, data)
-	defer closeResponseBody(ctx, res)
 	if err != nil {
 		return err
```
Comment (ChrysmOre) [inline cantabular/queries.go:279]: You need to expand this error to include the logData etc.. Pretty sure you did this before?

Comment (saminahbab) [reply cantabular/queries.go:279]: I have not redefined any errors, it was like that before. I'll expand however, one second

Comment (saminahbab) [reply cantabular/queries.go]: Done

File: cantabular/queries.go [hunk 3]
```diff
@@ -268,31 +268,36 @@ func (data *QueryData) Encode(query string) (bytes.Buffer, error) {
...
+		"query_data": data,
+	}
+
 	res, err := c.postQuery(ctx, graphQLQuery, data)
-	defer closeResponseBody(ctx, res)
 	if err != nil {
-		return err
+		return dperrors.New(
+			fmt.Errorf("failed to post query: %s", err),
+			res.StatusCode,
```
Comment (ChrysmOre) [inline cantabular/queries.go:281]: should be statusCode(err)

File: cantabular/dimensions_test.go [hunk 4]
```diff
@@ -68,8 +68,12 @@ func TestGetDimensionsUnhappy(t *testing.T) {
 		Convey("When GetDimensions is called", func() {
 			resp, err := cantabularClient.GetDimensions(testCtx, "Teaching-Dataset")
 
+			// Convey("Then the expected error is returned", func() {
```
Comment (ChrysmOre) [inline cantabular/dimensions_test.go:71]: Lots of tests still commented out, you probably want to replace with `So(err, ShouldNotBeNil)` in addition to the statusCode check you've added for these

Comment (saminahbab) [reply cantabular/dimensions_test.go:71]: bear with me on this one, ive been working on this as we have been pair debugging the coverage ticket, and just experimenting with the broken tests just to get them working, ill ping you when its cleaned up

**Review by PauloPortugal — APPROVED** [review]

**Pull request merged**